src/libsyntax/parse/parser.rs

```diff
@@ -644,6 +644,22 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn check_contextual_keyword(&mut self, ident: Ident) -> bool {
+        let tok = token::Ident(ident, token::Plain);
+        let r = self.token == tok;
```

Comment:
This is probably wrong, see the comment for the impl of `Eq` for `Ident` (ast.rs:104). You probably want non-hygenic comparison here.